project/hash.go

```diff
@@ -11,13 +11,11 @@ import (
 
 // GetServiceHash computes and returns a hash that will identify a service.
 // This hash will be then used to detect if the service definition/configuration
-// have changed and needs to be rebuild.
-func GetServiceHash(service Service) string {
+// have changed and needs to be recreate.
+func GetServiceHash(name string, config *ServiceConfig) string {
 	hash := sha1.New()
 
-	io.WriteString(hash, fmt.Sprintln(service.Name()))
-
-	config := service.Config()
+	io.WriteString(hash, name)
```

Comment:
@imikushin Okay, I figure out what's going on there.  I screwed up.  My original PR had the wrong commit, it wasn't the same logic we are using in RancherOS.  I've cherry picked the right commit and updated #16.  Sorry about that, can you rebase with #16?